src/bytes.rs

```diff
@@ -930,18 +930,18 @@ unsafe fn promotable_odd_drop(data: &mut AtomicPtr<()>, ptr: *const u8, len: usi
         let kind = shared as usize & KIND_MASK;
 
         if kind == KIND_ARC {
-            release_shared(shared as *mut Shared);
+            release_shared(shared.cast());
         } else {
             debug_assert_eq!(kind, KIND_VEC);
 
-            drop(rebuild_boxed_slice(shared as *mut u8, ptr, len));
+            drop(rebuild_boxed_slice(shared.cast(), ptr, len));
         }
     });
 }
 
 unsafe fn rebuild_boxed_slice(buf: *mut u8, offset: *const u8, len: usize) -> Box<[u8]> {
     let cap = (offset as usize - buf as usize) + len;
-    Box::from_raw(slice::from_raw_parts_mut(buf, cap))
+    Box::from_raw(ptr::slice_from_raw_parts_mut(buf, cap))
```

Comment:
looks like we'll need to bump MSRV to 1.42 for this?

Reply:
Using the slice version has issues re stacked borrows. I'll replace it with something else entirely.